### PR TITLE
Navnoor TLI : Added test cases for backend app.py

### PR DESCRIPTION
**Pull request by Navnoorsinghmahal**
### 📝 **Added Test Cases for Backend `app.py`** 🧪

This pull request introduces a series of unit tests for the backend application (`app.py`). The tests focus on ensuring the reliability of different routes and functionalities, including user registration, login, password change, and data uploads. Here's a breakdown of the updates:

### 🔍 **Test Cases Added**:
- **Headers Validation** 🛠️:
  - Test for missing `curr_user` and `table_name` in `/api/headers`.
  
- **Signup Endpoint** 📝:
  - Handling missing password in signup.
  - Successful user registration.
  - Value error for invalid email format.
  - Generic exception handling during signup.

- **Login Endpoint** 🔑:
  - Missing email/password validation.
  - Successful login.
  - Failure cases with wrong credentials.

- **Logout Endpoint** 🚪:
  - Ensures logout is successful and returns appropriate messages.

- **Data Upload Validation** 📂:
  - Test for missing CSV file and `curr_user` during data upload.

- **Model Upload** 🗂️:
  - Validation for missing data and invalid file formats during model upload.

- **Password Change** 🔐:
  - Handling missing fields in the password change request.
  - Mismatched passwords scenario.
  - Successful and failed password change attempts.

- **Share Endpoint** 📤:
  - Successful sharing of data and generated graph with mock data.
  - Handling invalid data in share requests.

- **Main Block Test** 🔴:
  - Verifying if the app runs with the correct configuration.

- **Home Route** 🏠:
  - Verifies the `/` route returns a 200 status with the correct message.

- **Email Fetching** 📧:
  - Test for scenarios where no user is logged in and fetches email correctly.

### CODE REVIEW

File: backend/tests/controllers/test_app.py [hunk 1]
```diff
@@ -0,0 +1,415 @@
...
+
+        encoded_data = "invalid_encoded_data"
+        response = self.client.get(f"/share/{encoded_data}", headers=self.headers)
+
+        self.assertEqual(response.status_code, 400)
+        self.assertIn("error", response.json)
+        self.assertEqual(response.json["error"], "Failed to decode or parse data")
+        self.assertEqual(response.json["message"], "Failed to decode encoded data")
+
+    class MainBlockTestCase(unittest.TestCase):
```
Comment (Jai0212) [inline backend/tests/controllers/test_app.py:380]: Whats the difference between this class and the previous class?

File: backend/tests/controllers/test_app.py [hunk 2]
```diff
@@ -0,0 +1,415 @@
+import unittest
```
Comment (Jai0212) [inline backend/tests/controllers/test_app.py:1]: How much percent test coverage is this?

**Review by Jai0212 — COMMENTED** [review]
Great work!

File: backend/tests/controllers/test_app.py (hunk 2 above)
Comment (Navnoorsinghmahal) [reply backend/tests/controllers/test_app.py:1]: its around 52%  right now and overall it is 58%

**Pull request merged**